Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Forge.DataContracts;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+
+namespace Microsoft.Forge.TreeWalker
+{
+    public class TreeSchemaValidator : ITreeSchemaValidator
+    {
+        private JSchema ForgeSchemaValidationRules;
+        public TreeSchemaValidator() => ForgeSchemaValidationRules = GetRulesFromPath("contracts\\ForgeSchemaValidationRules.json");
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemas(IList<ForgeTree> schemas)
```

Comment:
>IList<ForgeTree> schemas [](http://example.com/codeflow?start=70&length=24)

Instead of a list of ForgeTree, let's accept a Dictionary<string, ForgeTree>. That is the recommended way to handle multiple trees.